src/renderers/shared/fiber/ReactFiberExpirationTime.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @providesModule ReactFiberExpirationTime
+ * @flow
+ */
+
+'use strict';
+
+import type {PriorityLevel} from 'ReactPriorityLevel';
+const {
+  NoWork,
+  SynchronousPriority,
+  TaskPriority,
+  HighPriority,
+  LowPriority,
+  OffscreenPriority,
+} = require('ReactPriorityLevel');
+
+const invariant = require('fbjs/lib/invariant');
+
+// TODO: Use an opaque type once ESLint et al support the syntax
+export type ExpirationTime = number;
+
+const Done = 0;
+const Sync = 1;
+const Task = 2;
+const Never = Number.MAX_SAFE_INTEGER;
```

Comment:
This number is way too high. Since VMs (really mostly just V8) tries to optimize for integer types when they can, sometimes they want them to fit into a single 32bit word on 32-bit systems. This number might cause them to deopt that assumption and have to expand the slot.

It should be 2^30 or whatever is the highest we can fit in 31-bit.